common/src/main/java/com/microsoft/identity/common/internal/providers/azureactivedirectory/AzureActiveDirectoryOAuth2Strategy.java

```diff
@@ -1,20 +1,38 @@
 package com.microsoft.identity.common.internal.providers.azureactivedirectory;
 
+import android.net.Uri;
+
 import com.microsoft.identity.common.Account;
 import com.microsoft.identity.common.internal.providers.oauth2.AccessToken;
 import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationRequest;
 import com.microsoft.identity.common.internal.providers.oauth2.IDToken;
+import com.microsoft.identity.common.internal.providers.oauth2.OAuth2Configuration;
 import com.microsoft.identity.common.internal.providers.oauth2.OAuth2Strategy;
 import com.microsoft.identity.common.internal.providers.oauth2.RefreshToken;
 import com.microsoft.identity.common.internal.providers.oauth2.TokenRequest;
 import com.microsoft.identity.common.internal.providers.oauth2.TokenResponse;
 
+import java.net.URL;
+
 
 /**
  * The Azure Active Directory oAuth2 Strategy
  */
 public class AzureActiveDirectoryOAuth2Strategy extends OAuth2Strategy {
 
+    private AzureActiveDirectoryOAuth2Configuration mConfig = null;
+
+    public AzureActiveDirectoryOAuth2Strategy(OAuth2Configuration config) {
```

Comment:
At a glance, we should be able to get away with specifying the constructor parameter on `AzureActiveDirectoryOAuth2Strategy ` as `AzureActiveDirectoryOAuth2Configuration` so long as it extends `OAuth2Configuration` -- would still let you call `super()` and avoids the `instanceof` check